pkg/internal/apis/config/validate.go

```diff
@@ -130,6 +140,68 @@ func (n *Node) Validate() error {
 	return nil
 }
 
+func formatPortMapping(addr net.IP, port int, protocol PortMappingProtocol) string {
+	return fmt.Sprintf("%s/%s",
+		net.JoinHostPort(addr.String(), strconv.Itoa(port)),
+		protocol,
+	)
+}
+
+func validatePortMapping(portMappingsSet sets.String, addrSet sets.String, mapping PortMapping) error {
+	wildcardAddrIPv4 := net.ParseIP("0.0.0.0")
+	wildcardAddrIPv6 := net.ParseIP("::")
```

Comment:
for goalng is the same, so you can simplify into one unique variable `wildcardAdd` , left a comment below with a reference to a golang issue explaining it

Reply:
keeping both the wild card addrs as `net.ParseIP("0.0.0.0").Equal(net.ParseIP("::"))` will return false in golang and the user might provide port mapping for both under the assumption that they won't overlap with each other and will bind to different IP protocols

Reply:
yes, but for your validation porpuse see suggestion here https://github.com/kubernetes-sigs/kind/pull/3175#discussion_r1181035115

if you translate wildcard addresses to a constant and you build a map indexed by port/protocol, you must have only one wildecard in the value of the map, otherwise it means you'll have a collision problem